src/main/haskell/kore/src/SMT.hs

```diff
@@ -191,11 +203,27 @@ runSMT config SMT { getSMT } = do
 The declared name is returned as an expression for convenience.
 
  -}
+
+-- Need to quote every identifier in SMT between pipes
+-- to escape special chars
+escapeId :: Text -> Text
+escapeId name = "|" <> name <> "|"
+
 declare :: MonadSMT m => Text -> SExpr -> m SExpr
 declare name typ =
     liftSMT $ withSolver $ \solver ->
         SimpleSMT.declare solver name typ
 
+declareFun :: MonadSMT m => Text -> [SExpr] -> SExpr -> m SExpr
```

Comment:
Please document exported symbols (here and below). I'm not sure if above you intended that comment to work as documentation (probably not), but you should also make sure that `escapeId` has documentation.